web/src/app/components/job-list-item/job-list-item.component.ts

```diff
@@ -40,19 +131,34 @@ export class JobListItemComponent implements OnInit, OnDestroy {
   jobPinUrl: SafeUrl;
   readonly jobListItemActionsType = JobListItemActionsType;
   subscription: Subscription = new Subscription();
+  loiData = new Map<string, string[]>([]);
+  treeControl: FlatTreeControl<DynamicFlatNode>;
+  dataSource: DynamicDataSource;
+
+  getLevel = (node: DynamicFlatNode) => node.level;
+  isExpandable = (node: DynamicFlatNode) => node.expandable;
+  hasChild = (_: number, _nodeData: DynamicFlatNode) => _nodeData.expandable;
+
+  // Properties in this order are used to give a LOI a name, otherwise it will
+  // be given a default name
+  _inferredNamesProperties = ['label', 'caption', 'name', 'id', 'code', 'key'];
```

Comment:
Interpreting the properties map is quasi-business logic that will be used in several places. Can we move it into `LocalOfInterestService`?